src/components/Firealerts/Form/FirealertsForm.tsx

```diff
@@ -0,0 +1,142 @@
+import { FirealertsType, alertConfiguration, generateCloudEventWithData } from "./EventConfigs";
+import { Typography } from "@rmwc/typography";
+import { GridCell, GridRow } from "@rmwc/grid";
+import { Card } from "@rmwc/card";
+import { Callout } from "../../common/Callout";
+import React, { ChangeEvent, useEffect } from "react";
+import { Button } from "@rmwc/button";
+import { useEmulatorConfig } from "../../common/EmulatorConfigProvider";
+import FirealertsLog from "./FirealertsLog";
+import { useFirealerts } from "../api/useFirealerts";
+import { FirealertsTrigger } from "../models";
+import { Select } from "@rmwc/select";
+import { ZeroState } from "./ZeroState";
+
+import "./FirealertsForm.scss";
+
+export const FirealertsForm = () => {
+    const triggers = useFirealerts();
+    const implementedAlerts = getAlertsForTriggers(triggers);
+    const alertsList = Object.keys(implementedAlerts) as FirealertsType[];
+    const config = useEmulatorConfig('eventarc');
+
+    const [selectedAlert, setSelectedAlert] = React.useState(alertsList[0]);
+    const [alertData, setAlertData] = React.useState(alertConfiguration[selectedAlert]?.default);
+
+    useEffect(() => {
+        setAlertData(alertConfiguration[selectedAlert]?.default);
+    }, [selectedAlert]);
+
+
+
+    const updateData = (key: string, value: any) => {
+        key = key.slice(1);
+        const newAlertData = { ...alertData };
+        updateFromStringPath(newAlertData, key, value);
+        setAlertData(newAlertData);
+    }
+
+    
+    const sendAlert = async () => {
+        const event = generateCloudEventWithData(selectedAlert, alertData);
+        const payload = { events: [event] }
+        const url = `//${config.hostAndPort}/google/publishEvents`
+        console.log("Posting event to: " + url);
+        const res = await fetch(url, {
+            method: "POST",
+            body: JSON.stringify(payload)
+        });
+        console.log(res); // TODO(gburroughs): Add notification of success/error
+    }
+
+    const getTriggerName = (trigger: FirealertsTrigger): string => {
+        const tokens = trigger.triggerName.split("-");
+        return tokens[tokens.length - 2];
+    }
+
+    if (!triggers || triggers.length === 0) {
+        return (
+            <GridCell span={12}>
+                <ZeroState />
+            </GridCell>
+        );
+    }
+    return (
+        <>
+            <GridCell span={12}>
+                {!triggers || triggers.length === 0 ? <ZeroState /> :
+                    <GridRow>
+                        <GridCell span={12}>
+                            <Typography use="headline6">Send Test {alertConfiguration[selectedAlert].name} Alert</Typography>
+                        </GridCell>
+                        <GridCell span={12}>
+                            <Callout>For more information on what the fields mean see <a href={alertConfiguration[selectedAlert].link}>here</a></Callout>
```

Comment:
nit: Consider asking a tech writer to chime in on these sentences

Reply:
Thanks for the suggestion! I will look into it.